src/Rating.js

```diff
@@ -1,8 +1,228 @@
 'use strict';
 
 import RatingBase from './Rating.soy';
+import { core } from 'metal';
 
 class Rating extends RatingBase {
+    /**
+     * @inheritDoc
+     */
+    constructor(opt_config) {
+        super(opt_config);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    attached() {
+        this.delegate('mouseover', 'a', this.handleMouseOverEvent.bind(this));
+        this.delegate('click', 'a', this.handleClickEvent.bind(this));
+        this.on('mouseout', this.handleMouseOutEvent.bind(this));
+    }
+
+    /**
+     * Handles click event
+     * @param {Event} event
+     * @protected
+     */
+    handleClickEvent(event) {
+        if (!this.disabled) {
+            let index = parseInt(event.target.dataset.index, 10);
+
+            if (this.selectedIndex === index && this.canReset) {
+                this.reset();
+            }
+            else {
+                this.selectedIndex = index;
+            }
+        }
+    }
+
+    /**
+     * Reset rating attributes to its initial value
+     * @protected
+     */
+    reset() {
+        this.rate = null;
+        this.selectedIndex = -1;
+    }
+
+    /**
+     * Handles mouseout event
+     * @protected
+     */
+    handleMouseOutEvent() {
+        this.currentMouseTarget = undefined;
+        this.highlightStarIndex = this.selectedIndex;
+    }
+
+    /**
+     * Handles mouseover event
+     * @param {event} event
+     * @protected
+     */
+    handleMouseOverEvent(event) {
+        if (!this.disabled) {
+            let index = Number.parseInt(event.target.dataset.index, 10);
+
+            if (this.currentMouseTarget !== index) {
+                this.highlightStarIndex = index;
+            }
+
+            this.currentMouseTarget = index;
+        }
+    }
+
+    /**
+     * Syncronize seletected item attribute chagens
+     * @param {object} item
+     * @protected
+     */
+    syncSelectedItem(item) {
+        if (item) {
+            this.rate = item.value;
+        }
+    }
+
+    /**
+     * Current item selected index
+     * @param {number} index
+     * @protected
+     */
+    syncSelectedIndex(index) {
+        this.highlightStarIndex = index;
+        this.selectedItem = this.options[index] || null;
+    }
 }
 
+Rating.ATTRS = {
+
+    /**
+     * Block or unblock rating functionality
+     * @type {?boolean}
+     * @default false
+     */
+    cssClasses: {
+        value: {
+            off: 'glyphicon glyphicon-star-empty',
+            on: 'glyphicon glyphicon-star'
+        }
+    },
+
+    /**
+    * Block or unblock rating functionality
+    * @type {?boolean}
+    * @default false
+    */
+    disabled: {
+        value: false,
+        validator: core.isBoolean
+    },
+
+    /**
+    * The current rate
+    * @type {?number}
+    * @default 0
+    */
+    rate: {
+        value: null
+    },
+
+    /**
+    * The current highlighted index
```

Comment:
The description for this attribute is the same as `highlightStarIndex`, so I'm not sure what the difference between them is.